addons/l10n_pk_edi/models/account_move.py

```diff
@@ -0,0 +1,494 @@
+                title=error['title'], msg=error['msg']
+            ))
+        return super()._post(soft)
+
+    def action_l10n_pk_edi_send(self):
+        """
+            Create a attachment and Send the invoice to the FBR.
+        """
+        self.ensure_one()
+        attachment_vals = self._l10n_pk_edi_get_attachment_values()
+        self.env['ir.attachment'].create(attachment_vals)
+        self.invalidate_recordset(fnames=[
+            'l10n_pk_edi_attachment_id',
+            'l10n_pk_edi_attachment_file'
+        ])
+        self.message_post(attachment_ids=self.l10n_pk_edi_attachment_id.ids)
+        self._l10n_pk_edi_send({self: attachment_vals})
+        self.is_move_sent = True
+
+    # -------------------------------------------------------------------------
+    # E-Invocing(FBR) Eligibility
+    # -------------------------------------------------------------------------
+
+    def _get_l10n_pk_sales_groups(self):
+        return self.env['account.tax.group'].browse([
+            sale_group_id
+            for ref in (
+                'tax_group_pk_0',
+                'tax_group_pk_1',
+                'tax_group_pk_2',
+                'tax_group_pk_3',
+                'tax_group_pk_5',
+                'tax_group_pk_8',
+                'tax_group_pk_10',
+                'tax_group_pk_13',
+                'tax_group_pk_15',
+                'tax_group_pk_16',
+                'tax_group_pk_17',
+                'tax_group_pk_195',
+            )
+            if (sale_group_id := self.env['ir.model.data']._xmlid_to_res_model_res_id(
+                f"account.{self.company_id.id}_{ref}"
+            )[1])
+        ])
+
+    def _get_l10n_pk_withhold_groups(self):
+        return self.env['account.chart.template'].with_company(self.company_id).ref('tax_group_pk_wt')
+
+    def _l10n_pk_edi_eligible_for_fbr(self):
+        self.ensure_one()
+        # Check whether the invoice is eligible for E-Invocing(FBR) or not.
+        is_taxable = (
+            self._get_l10n_pk_sales_groups()
+            & self.line_ids.tax_ids.tax_group_id
+        )
+        return (
+            self.company_id.account_fiscal_country_id.code == 'PK'
+            and self.is_sale_document
+            and self.pos_order_ids
+            and is_taxable
+        )
+
+    def _l10n_pk_edi_get_default_enable(self):
+        self.ensure_one()
+        return (
+            self.state == 'posted'
+            and self.l10n_pk_edi_state != 'sent'
+            and self._l10n_pk_edi_eligible_for_fbr()
+        )
+
+    # -------------------------------------------------------------------------
+    # E-Invocing Methods
+    # -------------------------------------------------------------------------
+
+    def _l10n_pk_validate_moves(self):
+        corrupt_move_name = {}
+        move_ids = [move for move in self if move.is_sale_document()]
+        for move in move_ids:
+            if err_msg := move._l10n_pk_validate_move_lines():
```

Comment:
`_ids` post fix in variable names must be reserved for a list of ids or o2m field in model.
```suggestion
        move_ids = self.filtered(lambda m: m.is_sale_document())
        for move in move_ids:
            if err_msg := move._l10n_pk_validate_move_lines():
```
using two `for loops` (L174 and L183) seems abit weird can you please make it a bit simple? 😄